### PR TITLE
[FLINK-14745] Add dependencies of job as list of URLs in config.

### PR DESCRIPTION
**Pull request by kl0u**
## What is the purpose of the change

Currently, the PackagedProgram has methods that allow the "flattening" of the job jar, in case it contains other jars. This issue will allow to put the list of URLs in the configuration directly and independently from the PackagedProgram, so that then they can be picked up by the Executors.

## Verifying this change

This change is already covered by existing tests and it should also be tested during execution of the end-to-end tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-clients/src/main/java/org/apache/flink/client/cli/ExecutionConfigAccessor.java [hunk 1]
```diff
@@ -61,7 +62,10 @@ public static ExecutionConfigAccessor fromProgramOptions(final ProgramOptions op
 
 		final Configuration configuration = new Configuration();
 
-		configuration.setInteger(CoreOptions.DEFAULT_PARALLELISM, options.getParallelism());
+		if (options.getParallelism() != ExecutionConfig.PARALLELISM_DEFAULT) {
```
Comment (TisonKun) [inline flink-clients/src/main/java/org/apache/flink/client/cli/ExecutionConfigAccessor.java]: Does this commit go into a separated story?

Comment (kl0u) [reply flink-clients/src/main/java/org/apache/flink/client/cli/ExecutionConfigAccessor.java]: I will squash them when merging. It was just for the reviewing.

**Comment by kl0u** [comment]
Thanks for the review @TisonKun ! I will merge.

**Comment by kl0u** [comment]
Merged!

**Pull request closed without merging**